### ISSUE TITLE
Allow for specifying both <name> and <binding> / <converter> in <forcedType>

### ISSUE THREAD

**Issue by brettkail-wk** [jOOQ/jOOQ#13295]
### Use case:
We have a MySQL table with a `BIT(64)` column.  As a workaround for #10323, we're using `.withName("BIGINT")`, which produces expected output:
```
public final TableField<TestRecord, Long> TEST_FIELD = createField(DSL.name("test_field"), SQLDataType.BIGINT.nullable(false), this, "");
```
We would like to use a custom Java type to improve type safety, so we tried adding `.withUserType` and `.withConverter`, but we get this error:
```
Bad configuration for <forcedType/>. <userType/> is not allowed when <name/> is provided
```
...with this output:
```
public final TableField<TestRecord, TestValueSet> TEST_FIELD = createField(DSL.name("test_field"), SQLDataType.BIT.nullable(false), this, "", new TestValueSetConverter());
```
This has the desired `Field` type and converter, but the `SQLDataType` is wrong, so it does not compile because the converter implements `Converter<Long, TestValueSet>` rather than `Boolean` required by `SQLDataType.BIT`.

### Possible solution you'd like to see:

Allow specifying both the SQLDataType and user type with converter.

### Possible workarounds:

Continue to use `.withName` only.

### Versions:

- jOOQ: 3.16.4
- Java: 11
- Database (include vendor): MySQL
- OS: MacOS
- JDBC Driver (include name if inofficial driver): MySQL

**Comment by lukaseder** [comment]
Thanks a lot for your suggestion. I believe this is a duplicate, but I can't find the other issue, so let's just keep this one open and see which one gets implemented first.

I agree, it should be possible to combine the data type rewriting and user type features of the `forcedType` configuration. One shouldn't exclude the other.

The workaround is to implement a `Binding` and ignore the wrong `T` type of the binding. You don't *have to* use the `Binding::converter` from within your `Binding` even if you have to declare it.

**Comment by lukaseder** [comment]
Related issue: https://github.com/jOOQ/jOOQ/issues/15843

**Comment by lukaseder** [comment]
It looks like this would be quite simple to implement, *and* it would help solve https://github.com/jOOQ/jOOQ/issues/15843. Looking into this for jOOQ 3.19, possibly with backports

**Comment by lukaseder** [comment]
The reason why things were implemented the way they were is this:

- https://github.com/jOOQ/jOOQ/issues/3704

There was room for accidents when a `<forcedType>`'s `<name>` matched both a `SQLDataType` and a `<customType>`. The latter has been deprecated long ago, in jOOQ 3.10:

- https://github.com/jOOQ/jOOQ/issues/5750

So, that original issue is unlikely to still appear.

**Comment by lukaseder** [comment]
Having said so, the logic around this is quite delicate, so a backport is unlikely.

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.19.0
